Framework/Geometry/src/Instrument/Detector.cpp

```diff
@@ -169,5 +170,13 @@ void Detector::registerContents(ComponentVisitor &componentVisitor) const {
   componentVisitor.registerDetector(*this);
 }
 
+bool Detector::hasDetectorInfo() const {
+  const IComponent *root = this;
+  while (this->m_parent)
```

Comment:
This looks suspect. Is this supposed to be `root->m_parent` or something similar?